src/aap_eda/services/activation/activation_manager.py

```diff
@@ -251,12 +251,7 @@ def _start_activation_instance(self):
             self._error_activation(msg)
             raise exceptions.ActivationStartError(msg) from exc
 
-        # update status
-        self.set_status(ActivationStatus.RUNNING)
-        with transaction.atomic():
-            self.set_latest_instance_status(ActivationStatus.RUNNING)
-            self._set_activation_pod_id(pod_id=container_id)
-        self._reset_failure_count()
+        self._set_activation_pod_id(pod_id=container_id)
```

Comment:
So, if understand correctly what are you doing is to create the activation and leave it in starting state and wait for next loop of the monitor to change it as running, I disagree with this approach for some reasons:

1. It is not efficient. I think "starting" state should be a transitory state for the start operation and it should end up always in a "running" or "failed | error" state in the same loop. 

2. There is no a explicit predictable timeout. The time we allow for ansible-rulebook to report the first heartbeat is the time between monitor loop executions, which is variable. 

3 RULEBOOK_LIVENESS_TIMEOUT_SECONDS is set to 10 minutes!!! This is pretty excessive as a timeout for a starting activation. Why RULEBOOK_LIVENESS_TIMEOUT_SECONDS and RULEBOOK_LIVENESS_CHECK_SECONDS have so high values is a different story. 

4. We don't handle the lack of heartbeat as a starting error, being handled as unresponsive activation which is misleading for the user because in that case, the activation never started. 

In my opinion: 
It is this point of the code when we should wait for the heartbeat and set the activation and activation_instance as running. We should have a new system setting to define the timeout to receive the first heartbeat with a reasonable value, let's say 15? 30 seconds? 
If we reach the timeout, we can continue with the failed policy with a message indicating the process didn't start in time.

Reply:
@bzwei While testing this feature have you seen what happens if you don't send matching events to ansible-rulebook. The Heartbeat is controlled in the settings RULEBOOK_LIVENESS_TIMEOUT_SECONDS so if the user decides to make this as 5 minutes we won't get any session stats for 5 minutes.

Reply:
A new commit is pushed with a new constant RULEBOOK_READINESS_TIMEOUT_SECONDS and messages distinguish the timeout is due to Liveness check or Readiness check

Reply:
@mkanoor @bzwei I didn't check it, I assumed that ansible-rulebook will send the heartbeat when it starts regardless of the matching events. Is this true?  If this is not case, we can not use the heartbeat to transition an activation from starting to running.

Reply:
@bzwei maybe is because you are using rulebooks that triggers a rule at the beginning?